project_setup.py

```diff
@@ -0,0 +1,240 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Helps setup a new project based on this template."""
+
+import itertools
+import os
+from pathlib import Path
+import subprocess
+from typing import Callable, Iterable, Optional
+
+TEMPLATE_PACKAGE_NAME = "package_name"
+TEMPLATE_REPO_URL = "https://github.com/ssec-jhu/base-template"
+MSG_GIT_URL_NOT_FOUND = "--Remote URL not found in .git/config--"
+PROMPT_REMOTE_REPO = "Enter the URL of the remote repository [{url_guess}]: "
+PROMPT_PKG_NAME = "Enter the package name [{name_guess}]: "
+UPDATED_PYPROJECT_TOML = "pyproject.toml [project.urls]"
+UPDATED_README_CICD = "Updated README.md -> CI/Security badges"
+UPDATED_README_RTD = "Updated README.md -> RTD badge"
+UPDATED_README_CODECOV = "Updated README.md -> codecov badge"
+UPDATED_FILE = "Replaced `{template_name}` with `{package_name}` in {file_path} ✅"
+SETUP_COMPLETE = "{package_name} setup complete! 🎉🎉🎉\n"
+
+def make_name_safe(name:str) -> str:
+    """Makes a string safe to use as a package name.
+
+    Args:
+        name (str): The string to make safe.
+
+    Returns:
+        str: The safe string.
+
+    """
+    return name.lower().replace(' ', '_').replace('-', '_')
+
+
+def valid_file_predicate(file_path:str) -> bool:
+    """Checks if a file is valid for modification.
+
+    Args:
+        file_path (str): The path to the file to check.
+
+    Returns:
+        bool: Whether the file is valid for modification.
+
+    """
+    not_in_git = not file_path.startswith("./.git/")
+    not_image =  (
+        not file_path.endswith(".png")
+        and not file_path.endswith(".jpg")
+        and not file_path.endswith(".jpeg")
+    )
+    not_binary = not file_path.endswith(".DS_Store")
+    not_this = not file_path.endswith("project_setup.py")
+
+    return not_in_git and not_image and not_binary and not_this
+
+
+def get_repo_url(
+    user_provided_repo_url:str = "",
+) -> str:
+    """Gets the URL of the remote repository.
+
+    Args:
+        user_provided_repo_url (str, optional): The URL of the remote repository. Defaults to "".
+
+    Returns:
+        str: The URL of the remote repository.
+
+    """
+    url_guess = subprocess.check_output(
+        ['git', 'remote', 'get-url', 'origin']
+    ).decode().strip()
+
+    return user_provided_repo_url or url_guess
+
+
+def get_package_name(
+    repo_url:str,
+    user_provided_package_name:str = "",
+) -> str:
+    """Gets the package name from the user.
+
+    Guesses the package name from the repo URL.
+
+    Args:
+        repo_url (str): The URL of the remote repository.
+        user_provided_package_name (str, optional): The package name provided by the user. Defaults to "".
+
+    Returns:
+        str: The package name.
+    """
+    guessed_name = make_name_safe(Path(repo_url).stem)
+    return user_provided_package_name or guessed_name
+
+
+def replace_file_contents(
+    new_str:str,
+    old_str:str,
```

Comment:
Would be better if these were passed as a single dict, that way the file only has to be read and written once, rather than per str replacement.